src/main/scala/firrtl/passes/memlib/ReplaceMemMacros.scala

```diff
@@ -1,28 +1,94 @@
+    defaultPortSeq(mem) :+ Field("data", Flip, flattenType(mem.dataType)))
+
+  private def wPortToBundle(mem: DefAnnotatedMemory) = BundleType(
+    (defaultPortSeq(mem) :+ Field("data", Default, mem.dataType)) ++ (mem.maskGran match {
+      case None => Nil
+      case Some(_) => Seq(Field("mask", Default, createMask(mem.dataType)))
+    })
+  )
+  private def wPortToFlattenBundle(mem: DefAnnotatedMemory) = BundleType(
+    (defaultPortSeq(mem) :+ Field("data", Default, flattenType(mem.dataType))) ++ (mem.maskGran match {
+      case None => Nil
+      case Some(_) if getFillWMask(mem) => Seq(Field("mask", Default, flattenType(mem.dataType)))
+      case Some(_) => Seq(Field("mask", Default, flattenType(createMask(mem.dataType))))
+    })
+  )
+  // TODO(shunshou): Don't use createMask???
+
+  private def rwPortToBundle(mem: DefAnnotatedMemory) = BundleType(
+    defaultPortSeq(mem) ++ Seq(
+      Field("wmode", Default, BoolType),
+      Field("wdata", Default, mem.dataType),
+      Field("rdata", Flip, mem.dataType)
+    ) ++ (mem.maskGran match {
+      case None => Nil
+      case Some(_) => Seq(Field("wmask", Default, createMask(mem.dataType)))
+    })
+  )
+  private def rwPortToFlattenBundle(mem: DefAnnotatedMemory) = BundleType(
+    defaultPortSeq(mem) ++ Seq(
+      Field("wmode", Default, BoolType),
+      Field("wdata", Default, flattenType(mem.dataType)),
+      Field("rdata", Flip, flattenType(mem.dataType))
+    ) ++ (mem.maskGran match {
+      case None => Nil
+      case Some(_) if (getFillWMask(mem)) => Seq(Field("wmask", Default, flattenType(mem.dataType)))
+      case Some(_) => Seq(Field("wmask", Default, flattenType(createMask(mem.dataType))))
+    })
+  )
+
+  def memToBundle(s: DefAnnotatedMemory) = BundleType(
+    s.readers.map(Field(_, Flip, rPortToBundle(s))) ++
+    s.writers.map(Field(_, Flip, wPortToBundle(s))) ++
+    s.readwriters.map(Field(_, Flip, rwPortToBundle(s))))
+  def memToFlattenBundle(s: DefAnnotatedMemory) = BundleType(
+    s.readers.map(Field(_, Flip, rPortToFlattenBundle(s))) ++
+    s.writers.map(Field(_, Flip, wPortToFlattenBundle(s))) ++
+    s.readwriters.map(Field(_, Flip, rwPortToFlattenBundle(s))))
+
+  /** Creates a wrapper module and external module to replace a candidate memory
+   *  The wrapper module has the same type as the memory it replaces
+   *  The external module
+   */
+  def createMemModule(m: DefAnnotatedMemory, wrapperName: String): Seq[DefModule] = {
     assert(m.dataType != UnknownType)
     val wrapperIoType = memToBundle(m)
     val wrapperIoPorts = wrapperIoType.fields map (f => Port(NoInfo, f.name, Input, f.tpe))
+    // Creates a type with the write/readwrite masks omitted if necessary
     val bbIoType = memToFlattenBundle(m)
     val bbIoPorts = bbIoType.fields map (f => Port(NoInfo, f.name, Input, f.tpe))
     val bbRef = createRef(m.name, bbIoType)
-    val hasMask = containsInfo(m.info, "maskGran")
+    val hasMask = !m.maskGran.isEmpty
```

Comment:
Ok I guess it does... but maybe you'd rather use "isDefined" instead of ! isEmpty